File: Day-01-Bit-Manipulation/10-Power-Set/solution.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);

    int arr[n];
    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    int total = 1 << n;

    for(int num = 0; num < total; num++) {
        printf("[");
        for(int i = 0; i < n; i++) {
            if(num & (1 << i))
                printf("%d ", arr[i]);
        }
        printf("] ");
    }
}

File: Day-01-Bit-Manipulation/10-Power-Set/solution.java
import java.util.*;
class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];

        for(int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        int total = 1 << n;

        for(int num = 0; num < total; num++) {
            System.out.print("[");
            for(int i = 0; i < n; i++) {
                if((num & (1 << i)) != 0)
                    System.out.print(arr[i] + " ");
            }
            System.out.print("] ");
        }
    }
}